src/unk_02054884.c

```diff
@@ -80,79 +77,73 @@ void sub_02054988 (Party * param0, int param1, int param2, u16 param3)
     Pokemon_ResetMoveSlot(Party_GetPokemonBySlotIndex(param0, param1), param3, param2);
 }
 
-int sub_020549A0 (Party * param0, u16 param1)
-{
-    int v0, v1;
-    Pokemon * v2;
+// In many of the functions below, C99-style iterator declaration doesn't match
 
-    v1 = Party_GetCurrentCount(param0);
+int Party_HasMonWithMove (Party *party, u16 moveID)
+{
+    int i;
+    int partyCount = Party_GetCurrentCount(party);
 
-    for (v0 = 0; v0 < v1; v0++) {
-        v2 = Party_GetPokemonBySlotIndex(param0, v0);
+    for (i = 0; i < partyCount; i++) {
+        Pokemon *mon = Party_GetPokemonBySlotIndex(party, i);
 
-        if (Pokemon_GetValue(v2, MON_DATA_IS_EGG, NULL) != 0) {
+        if (Pokemon_GetValue(mon, MON_DATA_IS_EGG, NULL) != 0) {
             continue;
         }
 
-        if ((Pokemon_GetValue(v2, MON_DATA_MOVE1, NULL) == param1) || (Pokemon_GetValue(v2, MON_DATA_MOVE2, NULL) == param1) || (Pokemon_GetValue(v2, MON_DATA_MOVE3, NULL) == param1) || (Pokemon_GetValue(v2, MON_DATA_MOVE4, NULL) == param1)) {
-            return v0;
-            break;
+        if (Pokemon_GetValue(mon, MON_DATA_MOVE1, NULL) == moveID 
+            || Pokemon_GetValue(mon, MON_DATA_MOVE2, NULL) == moveID
+            || Pokemon_GetValue(mon, MON_DATA_MOVE3, NULL) == moveID
+            || Pokemon_GetValue(mon, MON_DATA_MOVE4, NULL) == moveID) {
+            return i;
         }
     }
 
-    return 0xff;
+    return PARTY_SLOT_NONE;
 }
 
-int sub_02054A10 (const Party * param0)
+int Party_AliveMonsCount (const Party *party)
 {
-    int v0, v1, v2;
-    Pokemon * v3;
+    int i;
+    int partyCount = Party_GetCurrentCount(party);
+    int count = 0;
 
-    v1 = Party_GetCurrentCount(param0);
-    v2 = 0;
-
-    for (v0 = 0; v0 < v1; v0++) {
-        v3 = Party_GetPokemonBySlotIndex(param0, v0);
-
-        if (sub_02054884(v3)) {
-            v2++;
+    for (i = 0; i < partyCount; i++) {
+        if (Pokemon_CanBattle(Party_GetPokemonBySlotIndex(party, i))) {
+            count++;
         }
     }
 
-    return v2;
+    return count;
 }
 
-Pokemon * Party_FirstBattler (const Party * param0)
+Pokemon * Party_FirstBattler (const Party *party)
```

Comment:
I feel like the naming between this and `Party_FirstMon` is a little confusing; maybe:

- `Party_FirstBattler` -> `Party_FindFirstEligibleBattler`
- `Party_FirstMon` -> `Party_FindFirstHatchedMon`